pkg/reconciler/cluster.go

```diff
@@ -57,16 +59,43 @@ func syncCluster(
 		if !reflect.DeepEqual(cr.Status, oldStatus) {
 			// Write back the status. Don't exit this handler until we
 			// succeed (will block other handlers for this resource).
```

Comment:
this could conceivably never exit. is that ok?

Reply:
Yeah, or at least in the sense that I'm not sure there's much else we can reasonably do here... if we don't update the status on the object then there's no way we can correctly manage it in the future.

A couple of comments on that situation:
* This function not-returning will only block future handling of this specific cluster object. Other objects can still be handled.
* It would be possible (with the change in this PR) to get out of that situation by deleting the object.

There are other ways to handle this situation without tying up a goroutine, and if these seem attractive/necessary we could create an issue to pursue them. For example:
* We could set ourselves a reminder of "don't allow any more changes to this object", save the desired status object (in memory), and schedule a periodic timer to kick off a retry of the status update. But I'm not sure this is notably different from using time.Sleep.
* Or we could (perhaps after some retries) set ourselves a reminder of "don't allow any more changes to this object" and then just mark the object as permanently corrupted.